src/App.tsx

```diff
@@ -15,48 +16,114 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'alphabet',
+  Length = 'length',
+  Default = '',
+}
+
+interface SortParams {
+  sortField: string,
+  isReversed: boolean,
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed }: SortParams,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case (SortType.Alphabet):
+          return good1.localeCompare(good2);
+
+        case (SortType.Length):
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.Default);
+  const [isReversed, setIsReversed] = useState(false);
+  const hasChanges = (sortField || isReversed);
+  const goods = getPreparedGoods(goodsFromServer, { sortField, isReversed });
+
+  function reset() {
+    setSortField(SortType.Default);
+    setIsReversed(false);
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classnames(
+            'button',
+            'is-info',
+            { 'is-light': sortField !== SortType.Alphabet },
+          )}
+          onClick={() => setSortField(SortType.Alphabet)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classnames(
+            'button',
+            'is-success',
+            { 'is-light': sortField !== SortType.Length },
+          )}
+          onClick={() => setSortField(SortType.Length)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={classnames(
+            'button',
+            'is-warning',
+            { 'is-light': !isReversed },
+          )}
+          onClick={() => {
+            setIsReversed(!isReversed);
```

Comment:
```suggestion
            setIsReversed(isReversed => !isReversed);
```

When we change state depends on previous state better to use callback

Reply:
This way I got a mistake - ''isReversed' is already declared in the upper scope.eslint[@typescript-eslint/no-shadow]''
So, I'm going to use smth like 'setIsReversed(currentReversed => !currentReversed);' Hope, it'll fit also)